src/modules/product/product.service.ts

```diff
@@ -97,31 +197,68 @@ export class ProductService {
     const slug = productInfo.name ? convertToSlug(productUpdateData.name) : '';
 
     if (!categoryId){
-      productDetails.forEach(async productDetails => {
-        const {product, productDetailId,... productDetailsUpdate} = productDetails;
-        return await this.productDetailsService.update(productDetailId, productDetailsUpdate);
-      })
+      for (const productDetail of product.productDetails) {
+        await this.productDetailsService.remove(productDetail.id);
+      }
       const productUpdateInfo = { ...productInfo, slug };
   
       Object.assign(product, productUpdateInfo);
+
+      const productUpdated = await this.productRepository.save(product);
+
+      const productDetailsData = [];
+
+      for (const productDetail of productDetails) {
+        const productDetailData = {
+          ...productDetail,
+          product: productUpdated
+        };
+
+        productDetailsData.push(productDetailData);
+      }
+    
+      await Promise.all(
+        productDetailsData.map(productDetail => 
+          this.productDetailsService.create(productDetail)
+        )
+      );
   
-      return await this.productRepository.save(product);
+      return productUpdated;
     }
     else {
       const category = await this.categoryService.findOne(categoryId);
     
       if (!category) {
         throw new Error(ErrorCode.CATEGORY_NOT_FOUND);
       }
-      productDetails.forEach(async productDetails => {
-        const {product, productDetailId,... productDetailsUpdate} = productDetails;
-        return await this.productDetailsService.update(productDetailId, productDetailsUpdate);
-      })
+
+      for (const productDetail of product.productDetails) {
+        await this.productDetailsService.remove(productDetail.id);
+      }
       const productUpdateInfo = { ...productInfo, category, slug };
   
       Object.assign(product, productUpdateInfo);
-  
-      return await this.productRepository.save(product);
+
+      const productUpdated = await this.productRepository.save(product);
+
+      const productDetailsData = [];
+
+      for (const productDetail of productDetails) {
+        const productDetailData = {
+          ...productDetail,
+          product: productUpdated
+        };
+
+        productDetailsData.push(productDetailData);
+      }
+    
+      await Promise.all(
+        productDetailsData.map(productDetail => 
+          this.productDetailsService.create(productDetail)
+        )
+      );
+    
+      return productUpdated;
```

Comment:
const productUpdated = await this.productRepository.save(product);

      const productDetailsData = [];

      for (const productDetail of productDetails) {
        const productDetailData = {
          ...productDetail,
          product: productUpdated
        };

        productDetailsData.push(productDetailData);
      }

      await Promise.all(
        productDetailsData.map(productDetail => 
          this.productDetailsService.create(productDetail)
        )
      );

      return productUpdated;
      đoạn code này bị lặp lại, if hay else gì cũng chạy thì tách ra làm gì ?